### ISSUE TITLE
AADAuthenticationMethodPolicyAuthenticator, AADAuthenticationMethodPolicyTemporary and potentially the other AADAuthenticationMethodPolicy* components bug with individual user assigned to Authentication Policy

### ISSUE THREAD

**Issue by PierreColyn** [microsoft/Microsoft365DSC#3400]
When I run the above mentioned components (AADAuthenticationMethodPolicyAuthenticator, AADAuthenticationMethodPolicyTemporary) I get the following error if an individual user is assigned to the Authentication Method:

get-MgGroup : Resource 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' does not exist or one of its queried reference-property
objects are not present.
At C:\Users\#Username#\Documents\WindowsPowerShell\Modules\Microsoft365DSC\1.23.614.1\DSCResources\MSFT_AADAuthentication
MethodPolicyTemporary\MSFT_AADAuthenticationMethodPolicyTemporary.psm1:139 char:17

When an AAD group is assigned to the Authentication Method the script works well, but when the script uses the Get-MgGroup command on an individually assigned user the script fails as per above.

**Comment by PierreColyn** [comment]
Thank you for the response.

The script search for resources assigned to a specific Authentication Method.
For each resource assigned to the authentication method, the script calls a Get-MgGroup on the resource. I assume that it is to return the members of an AAD group (script assumes it is an AAD Group assigned to the resource.
The error occurs if the assigned resource is an individual AAD user - the script also calls Get-MgGroup on the user, which fails.

Proposed solution:
Check if the assigned resource is a group or a user and only call Get-MgGroup if the assigned resource is a group, else return the user.

**Comment by infinity-au** [comment]
I don't believe individual users are a valid input for AuthenticationMethodPolicies, it looks to me like only groups or all_users are valid options.

**Comment by PierreColyn** [comment]
Thank you, you are correct, this was previously possible but with the latest Azure updates Microsoft has removed the functionality. They however still allow existing individual users that were previously added
.
See this video: (at 47:09 / 1:00:45) https://www.youtube.com/watch?v=lajeFoCr2KM

![image](https://github.com/microsoft/Microsoft365DSC/assets/136440235/4d6012a6-39cd-406a-8d04-adbc6448b362)

**Comment by PierreColyn** [comment]
@andikrueger, thank you for the update. It is up to you to decide if you want to make the change or not. I've updated the one script locally and it is really a minor change, unless I'm missing something:

![image](https://github.com/microsoft/Microsoft365DSC/assets/136440235/9e72ffd1-493e-4c3e-866e-d2ee185e6d66)

**Comment by FabienTschanz** [comment]
Only groups are possible today. Closing this issue.
